event_session/models/event_session.py

```diff
@@ -0,0 +1,275 @@
+        """Determine reserved, available, reserved but unconfirmed and used
+        seats by session.
+        """
+        # initialize fields to 0
+        for session in self:
+            session.seats_unconfirmed = session.seats_reserved = \
+                session.seats_used = session.seats_available = 0
+        # aggregate registrations by event session and by state
+        if self.ids:
+            state_field = {
+                'draft': 'seats_unconfirmed',
+                'open': 'seats_reserved',
+                'done': 'seats_used',
+            }
+            query = """
+                SELECT session_id, state, count(session_id)
+                FROM event_registration
+                WHERE session_id IN %s AND state IN ('draft', 'open', 'done')
+                GROUP BY session_id, state """
+            self._cr.execute(query, (tuple(self.ids),))
+            for session_id, state, num in self._cr.fetchall():
+                session = self.browse(session_id)
+                session[state_field[state]] += num
+        # compute seats_available
+        for session in self:
+            if session.seats_max > 0:
+                session.seats_available = session.seats_max - (
+                    session.seats_reserved + session.seats_used)
+            session.seats_expected = (
+                session.seats_unconfirmed + session.seats_reserved +
+                session.seats_used)
+            session.seats_available_expected = (
+                session.seats_max - session.seats_expected)
+            session.seats_available_pc = (
+                session.seats_expected * 100 / float(session.seats_max))
+
+    @api.multi
+    @api.depends('date_tz', 'date_begin')
+    def _compute_date_begin_located(self):
+        for session in self:
+            if session.date_begin:
+                self_in_tz = session.with_context(
+                    tz=(session.date_tz or 'UTC')
+                )
+                date_begin = fields.Datetime.from_string(session.date_begin)
+                session.date_begin_located = fields.Datetime.to_string(
+                    fields.Datetime.context_timestamp(self_in_tz, date_begin),
+                )
+            else:
+                session.date_begin_located = False
+
+    @api.multi
+    @api.depends('date_tz', 'date_end')
+    def _compute_date_end_located(self):
+        for session in self:
+            if session.date_end:
+                self_in_tz = session.with_context(
+                    tz=(session.date_tz or 'UTC')
+                )
+                date_end = fields.Datetime.from_string(session.date_end)
+                session.date_end_located = fields.Datetime.to_string(
+                    fields.Datetime.context_timestamp(self_in_tz, date_end),
+                )
+            else:
+                session.date_end_located = False
+
+    @api.onchange('event_id')
+    def onchange_event_id(self):
+        self.seats_min = self.event_id.seats_min
```

Comment:
Use `self.update` for making only one write operation (in case this is called outside an UI onchange itself).